File: apps/dotnet/FizzBuzz/Domain/Type/FizzBuzzType03.cs
namespace FizzBuzz.Domain.Type;

using FizzBuzz.Domain.Model;

public class FizzBuzzType03 : IFizzBuzzType
{
    private const int FizzBuzzNumber = 15;

    public FizzBuzzValue Generate(int number)
    {
        var value = number % FizzBuzzNumber == 0 ? "FizzBuzz" : number.ToString();
        return new FizzBuzzValue(number, value);
    }
}

File: apps/typescript/src/fizzbuzz/domain/type/fizz-buzz-type-03.ts
import { FizzBuzzValue } from "../model/fizz-buzz-value.js";
import type { FizzBuzzType } from "./fizz-buzz-type.js";

export class FizzBuzzType03 implements FizzBuzzType {
  generate(number: number): FizzBuzzValue {
    if (number % 15 === 0) {
      return new FizzBuzzValue("FizzBuzz", number);
    }
    return new FizzBuzzValue(number.toString(), number);
  }
}
